packages/compass-global-writes/src/store/reducer.ts

```diff
@@ -441,6 +659,45 @@ const setNamespaceBeingSharded = (
       type: GlobalWritesActionTypes.SubmittingForShardingFinished,
       managedNamespace,
     });
+    dispatch(pollForShardKey());
+  };
+};
+
+const pollForShardKey = (): GlobalWritesThunkAction<
+  void,
+  NextPollingTimeoutSetAction
+> => {
+  return (dispatch, getState) => {
+    const { pollingTimeout, isPluginTitleVisible } = getState();
+    if (
+      !isPluginTitleVisible || // user is not in the Collection Workspace
+      pollingTimeout // prevent double polling
+    ) {
+      return;
+    }
+    const timeout = setTimeout(
+      () => {
+        void dispatch(fetchNamespaceShardKey());
+      },
+      process.env.NODE_ENV !== 'test' ? POLLING_INTERVAL : TEST_POLLING_INTERVAL
+    );
+
+    dispatch({
+      type: GlobalWritesActionTypes.NextPollingTimeoutSet,
+      timeout,
+    });
+  };
+};
+
+const stopPollingForShardKey = (): GlobalWritesThunkAction<
```

Comment:
Can you clarify why we're doing this? Is it because this is what's happening in the indexes? If so, I don't think the case is really the same and seems like we're complicating the flow here quite a lot without really the need for it. We're polling only until the data is ready, so we can honestly just continue to do it even if the tab is off screen because there is a very clear stopping point for this logic.

Reply:
Yes actually, I only started to think about this once I've heard about the indexes. It's true that it's more complicated and less useful in this case. If you think it's not worth it, I can clean it up

Reply:
I would have a preference for keeping the implementation simple if possible, yeah. If we don't have a strong reason for this, probably better to avoid it if this make sense to you (and sorry for only spotting this after the implementation was done)